src/cls/rbd/cls_rbd.cc

```diff
@@ -7601,12 +7601,6 @@ std::string image_id_from_key(const std::string &key) {
  */
 int trash_add(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
 {
-  int r = cls_cxx_create(hctx, false);
```

Comment:
> keep this for robustness.

... and backwards compatibility.  Otherwise, a user with an older librbd running against newer OSDs wouldn't be able to remove any images unless they already happen to have the trash object.

Reply:
Good catch about the compatibility